static/app/components/dropdownMenu/item.tsx

```diff
@@ -134,7 +134,7 @@ function BaseDropdownMenuItem(
 
     if (isHovered && isFocused) {
       if (isSubmenu) {
-        state.selectionManager.select(node.key);
+        state.selectionManager.replaceSelection(node.key);
```

Comment:
turns out `selectionManager.select()` _toggles_ the specified value, so if `node.key` is already selected it will be deselected.